code/workhorse/prepare_data.R

```diff
@@ -0,0 +1,85 @@
+set_res_path <- function(simulator) {
+  cols = names(simulator$setting)[c(1,2)]
+  values = unlist(simulator$setting)[c(1,2)]
+  folder = paste0(cols,values,sep="",collapse = "-")
+  dir.create(file.path(out_data_path,folder), showWarnings = FALSE)
+  simulator$res_path = file.path(out_data_path,folder,simulator$setting["method"])
+  cat("create folder")
+  cat(simulator$res_path)
+  dir.create(simulator$res_path,showWarnings = FALSE)
+}
+
+set_res_name <- function(simulator){
+  
+  simulator$res_name = paste("every",simulator$setting["add"],"iter_num",simulator$setting["iter_num"]
+                             ,sep = "_")
+}
+
+save_benchmark <- function(simulator) {
+  
+  benchmark_file = paste(simulator$res_name,"_bench.csv",sep="")
+  write.csv(simulator$benchmark,file.path(simulator$res_path,benchmark_file))
+}
+
+
+
+load_data <- function(simulator) {
+  # load and sort the data, the first 14 rows covers all the 
+
+  # gene_id = simulator$setting[["gene_id"]]
+  data = read.csv(simulator$input_path, stringsAsFactors=FALSE)
+  con_names = data[,1]
+  data = data[,-1]
+  once_rows = which(apply(data[,1:14],1,sum)==1)
+  
+  #you can use a 2-d table to image this
+  set.seed(simulator$setting[["random_seed"]]) # shuffle before splitting
+  antiseptics = sample(colnames(data)[1:10], 10, replace = FALSE)
+  antibiotics = sample(colnames(data)[11:14], 4, replace = FALSE)
+
+  row_index = sample(1:10, 10, replace = FALSE)
+  col_index = c(1,2,3,4,3,2,1,2,4,1)
```

Comment:
not clear why these numbers are chosen and hardcoded (similar comment applies to other numbers used in this file).